src/execution.rs

```diff
@@ -93,48 +57,8 @@ impl File {
             self.execute_strict_into(graph, tree, source, config, cancellation_flag)
         }
     }
-}
-
-impl File {
-    /// Executes this graph DSL file against a source file, saving the results into an existing
-    /// `Graph` instance.  You must provide the parsed syntax tree (`tree`) as well as the source
-    /// text that it was parsed from (`source`).  You also provide the set of functions and global
-    /// variables that are available during execution. This variant is useful when you need to
-    /// “pre-seed” the graph with some predefined nodes and/or edges before executing the DSL file.
-    fn execute_strict_into<'a, 'tree>(
-        &self,
-        graph: &mut Graph<'tree>,
-        tree: &'tree Tree,
-        source: &'tree str,
-        config: &mut ExecutionConfig,
-        cancellation_flag: &dyn CancellationFlag,
-    ) -> Result<(), ExecutionError> {
-        self.check_globals(config.globals)?;
-        let mut locals = VariableMap::new();
-        let mut scoped = ScopedVariables::new();
-        let current_regex_captures = Vec::new();
-        let mut function_parameters = Vec::new();
-        let mut cursor = QueryCursor::new();
-        for stanza in &self.stanzas {
-            cancellation_flag.check("executing stanza")?;
-            stanza.execute(
-                tree,
-                source,
-                graph,
-                config,
-                &mut locals,
-                &mut scoped,
-                &current_regex_captures,
-                &mut function_parameters,
-                &mut cursor,
-                &self.shorthands,
-                cancellation_flag,
-            )?;
-        }
-        Ok(())
-    }
 
-    pub(crate) fn check_globals(&self, globals: &Globals) -> Result<(), ExecutionError> {
+    pub(self) fn check_globals(&self, globals: &Globals) -> Result<(), ExecutionError> {
```

Comment:
TIL `pub(self)`!  Is that "this module and nested modules, but not any other modules in the crate"?

Reply:
Indeed! The visibility in Rust is quite nice like that. It gives you the possibility to split things in different modules without having to expose it to unrelated modules in the same crate.